lib/srv/uacc/uacc.h

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef UACC_C
+#define UACC_C
+
+#include <unistd.h>
+#include <stdint.h>
+#include <string.h>
+#include <utmp.h>
+#include <errno.h>
+
+int UACC_UTMP_MISSING_PERMISSIONS = 1;
+int UACC_UTMP_WRITE_ERROR = 2;
+int UACC_UTMP_READ_ERROR = 3;
+int UACC_UTMP_FAILED_OPEN = 4;
+
+// The max byte length of the C string representing the TTY name.
+static int max_len_tty_name() {
+    return UT_LINESIZE;
+}
+
+// Low level C function to add a new USER_PROCESS entry to the database.
+// This function does not perform any argument validation.
+static int uacc_add_utmp_entry(const char *username, const char *hostname, const int32_t remote_addr_v6[4], const char *tty_name, const char *id, int32_t tv_sec, int32_t tv_usec) {
+    struct utmp entry;
+    entry.ut_type = USER_PROCESS;
+    strcpy((char*) &entry.ut_line, tty_name);
+    strcpy((char*) &entry.ut_id, id);
```

Comment:
```c
strncpy((char*) &entry.ut_id, id, sizeof(entry.ut_id))
```

Reply:
We already perform these checks in Go side so they're not needed here.

Reply:
The only reason they'd be useful is if glibc suddenly changed these sizes. I guess I'll add them in, can't hurt.

Reply:
It's not just that - next security audit will flag them for using insecure APIs and we will have to change that anyways.